profitbricks-rest/src/main/java/org/apache/jclouds/profitbricks/rest/binder/BaseProfitBricksRequestBinder.java

```diff
@@ -16,20 +16,18 @@
  */
 package org.apache.jclouds.profitbricks.rest.binder;
 
-import static com.google.common.base.Preconditions.checkNotNull;
-
 import java.util.Map;
-
 import org.jclouds.http.HttpRequest;
 import org.jclouds.rest.MapBinder;
-
 import com.google.common.base.Strings;
-
 import org.jclouds.io.MutableContentMetadata;
 import org.jclouds.io.payloads.BaseMutableContentMetadata;
+import static com.google.common.base.Preconditions.checkNotNull;
 
 public abstract class BaseProfitBricksRequestBinder<T> implements MapBinder {
 
+   protected final String endpointPath = "/rest/v2";
```

Comment:
In fact that is not a good option, as when building the context, jclouds substitutes several values and properties from the metadata based on the properties passed to the context builder (it is not a use case right now here, but even when the provider metadata sets the endpoint users can override it when creating the context)., so you really need to use the injection approach to inject the right endpoint.

Please change the constructor back to inject the `@Provider Supplier<URI> endpointSupplier`. Regarding the test, I've had a look and there is no need to configure the `LocationModule`; just binding the supplier used in the base binder class is enough:

```java
Injector injector = Guice.createInjector(new GsonModule(), new AbstractModule() {
   @Override protected void configure() {
      bind(new TypeLiteral<Supplier<URI>>() {
      }).annotatedWith(Provider.class).toInstance(
         Suppliers.ofInstance(URI.create(new ProfitBricksProviderMetadata().getEndpoint())));
      }
   });
```
I've applied the injection approach and configured the `UpdateImageRequestBinderTest` like this and tests passed. Could you apply the same pattern to the rest of the tests?